plugins/uefi-capsule/fu-uefi-capsule-plugin.c

```diff
@@ -692,6 +712,30 @@ fu_uefi_capsule_plugin_get_default_esp(FuPlugin *plugin, GError **error)
 		g_debug("%s", str->str);
 	}
 
+	if (esp_volumes->len == 1) {
+		FuVolume *esp = g_ptr_array_index(esp_volumes, 0);
+		g_autoptr(FuDeviceLocker) locker = NULL;
+
+		/* ensure it can be mounted */
+		locker = fu_volume_locker(esp, error);
+		if (locker == NULL)
+			return NULL;
```

Comment:
I'm confused why we need the mounting...

Reply:
To get the mount point.  If someone specifies `/foo` in `fwupd.conf` but it's mounted at `/boot` from fstab/udisks, we should reject it.

Reply:
I think there was some subtle assumptions previously.  In the "simple" case of only one real ESP on the system, but user specified something different I think it mostly worked because we would use udisks to get the ESP.  Then when we added their path if it matched up then the deduplication rejected the one they added.  

The recent bug report was something that udisks rejected because it was the wrong partition type and it brought out the fact that we didn't do enough validation of the case of just one ESP.

Reply:
I've changed it so mounting only happens when validating, not even in the case of no EspLocation key.